src/lint/lint.ts

```diff
@@ -57,6 +58,29 @@ export async function lint(options: Options, config: ProjectConfig) {
   });
   const linter = new lintLib.Linter(rules, analyzer);
 
+  if (options.watch) {
+    return watchLoop(analyzer, linter, options, config, filter);
+  } else {
+    return runOnce(analyzer, linter, options, config, filter);
+  }
+}
+
+interface PrivateOptions extends Options {
+  /**
+   * When running in --watch mode we want to report warnings if we're running
+   * with --fix but there weren't any warnings to fix.
+   */
+  reportIfNoFix?: boolean;
+}
+
+async function runOnce(
```

Comment:
+docs! also I'd prefer `run` instead of `runOnce`, I think run being singular is implied, especially if there's a second method that is explicitly "multiple/loop/watch/etc"